lib/wor/batchifier.rb

```diff
@@ -12,16 +12,24 @@ module Batchifier
     # => optionaly a batch_size and a merge strategy. It will slice the collection and
     # => apply the chozen strategy to all chunks and merge the results. It expects the responses
     # => to be Hash. It can ignore them if the given strategy is no_response
-    def execute_in_batches(collection, batch_size: 100, strategy: :add)
+    def execute_in_batches(collection, batch_size: 100, strategy: :add, &block)
       strategy_class = classify_strategy(strategy)
       collection.lazy.each_slice(batch_size).inject(strategy_class.new.base_case) do |rec, chunk|
-        response = yield(chunk)
+        response = batch_process(chunk, &block) if block_given?
         merge(response, rec, strategy)
       end
     end
 
     private
 
+    def batch_process(chunk)
+      yield chunk
+    rescue StandardError => e
+      puts "Backtrace:\n\t#{e.backtrace.join("\n\t")}"
+      raise Wor::Batchifier::Exceptions::CustomStrategyMergingError,
+        "Custom strategy could not be merged"
```

Comment:
This isn't very accurate. I'd say that the custom merge strategy defined by the developer raised an exception during execution. Maybe we could rephrase it like:

"Custom Merge Strategy from #{ TheCustomStrategy.class.name } raised # { TheException.class.name } during execution.
Backtrace: ............. "

What you think about this?

Reply:
Ok, I will add this and the other suggested changes.